ssl/ssl_lib.c

```diff
@@ -2360,10 +2360,14 @@ int SSL_renegotiate_pending(const SSL *s)
 
 int SSL_new_session_ticket(SSL *s)
 {
-    if (SSL_in_init(s) || SSL_IS_FIRST_HANDSHAKE(s) || !s->server
+    /* If we are in init because we're sending tickets, okay to send more. */
+    if ((SSL_in_init(s) && s->ext.extra_tickets_expected == 0)
+            || SSL_IS_FIRST_HANDSHAKE(s) || !s->server
             || !SSL_IS_TLS13(s))
         return 0;
     s->ext.extra_tickets_expected++;
+    if (s->rlayer.wbuf[0].left == 0 && !SSL_in_init(s))
+        ossl_statem_set_in_init(s, 1);
```

Comment:
How does this impact non-QUIC?

Reply:
This is the same check for "are we at a record boundary" that's already used (for non-QUIC) at the start of `ssl3_write_bytes()` where we would otherwise be writing the tickets out.  The point of this commit is to evaluate the condition when `SSL_new_session_ticket()` is called instead of waiting until an application-data write to check.

Reply:
Just wanted to make sure nothing bad happened in that case!

Reply:
It's a good question to ask!
I believe it is fine, though, and the test coverage here is pretty good.